src/wallet/coinselection.cpp

```diff
@@ -317,6 +318,8 @@ std::vector<CInputCoin>::iterator OutputGroup::Discard(const CInputCoin& output)
     if (it == m_outputs.end()) return it;
     m_value -= output.txout.nValue;
     effective_value -= output.effective_value;
+    fee -= output.m_fee;
+    long_term_fee -= output.m_long_term_fee;
```

Comment:
Should the reverse not also be added to the `Insert` method above?

Reply:
Yes, but no. The `OutputGroup`s are created (and `Insert` is called) before the fees are set. But `Discard` is called after the fees are set. So not incrementing `fee` and `long_term_fee` in `Insert` is safe because they are 0 at that time.

Reply:
Yeah, I think it would be better if it was added. Otherwise, bugs could be introduced if `Insert` is used differently in the future.

Reply:
I've added it to insert.